### PR TITLE
Refactor structure

### LINKED ISSUE #1: Improve project structure
By comment of @maj24 the folders:

- `Data/Remote/Services`
- `Domain`

Are examples that uses the `hooks` defined by *YellowPod* and maybe we should move this folders outside inside an example project.

### PR DESCRIPTION
**Pull request by luisburgos**
This closes #1 

Major changes on:

* [Project structure](https://github.com/yellowme/base-ios/tree/refactor/structure/YellowPod)
* [How to Guides](https://github.com/yellowme/base-ios/tree/refactor/structure/YellowPod/YellowPod/Data/Local)
* README guides like:
  * [Extensions for Collections and Tables](https://github.com/yellowme/base-ios/tree/refactor/structure/YellowPod/YellowPod/Common/Extensions/Collections)
  * [Local data folder](https://github.com/yellowme/base-ios/tree/refactor/structure/YellowPod/YellowPod/Data/Local)
  * [Remote data folder](https://github.com/yellowme/base-ios/tree/refactor/structure/YellowPod/YellowPod/Data/Remote)

### CODE REVIEW

**Comment by codeams** [comment]
Are we following a specific Code Style Guide? If so, I think It's worth adding a reference to it. For both users and contributors who want to use, maintain or extend the Pod.

File: YellowPod/YellowPod/Common/Extensions/UIApplication+Ext.swift [hunk 1]
```diff
@@ -2,39 +2,22 @@
...
 extension UIApplication {
     class func topViewController(controller: UIViewController? = UIApplication.shared.keyWindow?.rootViewController) -> UIViewController? {
         if let navigationController = controller as? UINavigationController {
             return topViewController(controller: navigationController.visibleViewController)
         }
-        if let tabController = controller as? UITabBarController {
-            if let selected = tabController.selectedViewController {
-                return topViewController(controller: selected)
-            }
+        if let tabController = controller as? UITabBarController, let selected = tabController.selectedViewController  {
```
Comment (codeams) [inline YellowPod/YellowPod/Common/Extensions/UIApplication+Ext.swift]: This line seems too long to me. Would it be a good idea to split the if statement?

Maybe we should set a maximum permitted amount of chars per line.

**Comment by luisburgos** [comment]
@codeams We are following the [Raywenderlich Styleguide](https://github.com/raywenderlich/swift-style-guide) but we need to pick a subset of most important rules in the near future. I think it's a good idea to add the reference. I will make the changes.

File: YellowPod/YellowPod/Common/Extensions/UIApplication+Ext.swift (hunk 1 above)
Comment (luisburgos) [reply YellowPod/YellowPod/Common/Extensions/UIApplication+Ext.swift]: The style guide we are following does not have any rules related. I agree with your comment, maybe we could add this as a "code review recipe" in [our guides repo](https://github.com/yellowme/guides/blob/master/code-review/SWIFT.md)

Comment (luisburgos) [reply YellowPod/YellowPod/Common/Extensions/UIApplication+Ext.swift]: I just reviewed the new styleguide version:

> - Long lines should be wrapped at around 70 characters. **A hard limit is intentionally not specified.**
> - Avoid trailing whitespaces at the ends of lines.
> - Add a single newline character at the end of each file.

Unlike `a multiple guard` statement, the `multiple if let` statement does not looks pretty when you separate each `let` per line.

Comment (luisburgos) [reply YellowPod/YellowPod/Common/Extensions/UIApplication+Ext.swift]: This refactor reduces the line a bit:

```swift
if let selectedFromTabBar = (controller as? UITabBarController)?.selectedViewController {
    return topViewController(controller: selectedFromTabBar)
}
```

**Review by codeams — APPROVED** [review]

**Pull request merged**